tests/integration/istioctl/analyze/analyze_test.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package istioctl
+
+import (
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"istio.io/istio/galley/pkg/config/analysis/msg"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/istioctl"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/label"
+)
+
+const (
+	serviceRoleBindingFile = "testdata/servicerolebinding.yaml"
+	serviceRoleFile        = "testdata/servicerole.yaml"
+)
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite("istioctl_analyze_test", m).
+		Label(label.CustomSetup).
```

Comment:
this doesn't have a custom setup, as noted by `istio.Setup(nil, nil)`

Reply:
As such, please move it out of the `analyze` package. Counter to normal code organization, making it in a new package means it will do a full setup and teardown of Istio which is wasteful. This can just be in the istioctl package

Reply:
Fixed.